src/api/rif-marketplace-cache/cacheController.ts

```
@@ -1,26 +1,27 @@
-import { maplistingToType } from 'api/utils';
-import { MarketItemType, MarketListingTypes, MarketFilterIface } from 'models/Market';
 import feathers from '@feathersjs/feathers';
 import socketio from '@feathersjs/socketio-client';
+import { MarketFilter } from 'models/Market';
 import io from 'socket.io-client';
 
-// TODO: extract
 const CACHE_BASE_ADDR = process.env.REACT_APP_CACHE_ADDR || 'http://localhost:3030';
 
 const socket = io(CACHE_BASE_ADDR);
 const client = feathers();
 client.configure(socketio(socket));
 // <- extract
 
-const services = {
-    offers: client.service(`rns/v0/offers`)
+const service = {
+    current: null,
+}
+export const createService = (path: string) => {
+    service.current = client.service(path);
+    return path;
 }
 
-
-export const fetchMarketDataFor = async (listingType: MarketListingTypes, filters?: MarketFilterIface):
-    Promise<MarketItemType[]> => {
-    const data = await services.offers.find({
+export const fetchMarketData = async (filters?: MarketFilter) => {
+    if (!service.current) throw Error('Not connected to a service');
+    return await (service.current as any).find({
         query: filters
     })
-    return maplistingToType(data.data, listingType);
-}
\ No newline at end of file
+}
```

I really like this approach, but was wondering if it would be better to get the service object from a function that already checks if the service was created so we don't have to manually check it every time we want to use it. Something like `getService()` and if it's not created we can throw the error there, otherwise we return the service object.
What do you think?

---

I just wanted to keep it simple, really.
I'm not sure what you mean, but I can only think of one use case when the appropriate service is already created and that is when you navigate to domains, then to landing and back to domains (or any other service, but only the same one as just previously visited). Otherwise it's always a different service.

---

If I had one service creator then I'd have to pass that service around to each fetch function I'd talk to. This way (also I thought) I keep the respective domains (as in business domains not RNS domains.] ) separated.